### PR TITLE
Feature/fm 38 common api for location data

### PR DESCRIPTION
**Pull request by tariqhamid**
please review this change

the rake task is called from the command line,
rake db:static

For completeness, the command,
rake db:setup
also loads the static data

Look Mum no ActiveRecord !

### CODE REVIEW

File: app/models/concerns/static_record.rb [hunk 1]
```diff
@@ -32,5 +32,21 @@ def where(arg)
...
+    # factory method
+    # each row of data in the database is used to create a new object
+    def load_db(query)
+      res = ActiveRecord::Base.connection_pool.with_connection { |con| con.exec_query(query) }
+      @all = []
+      res.each do |row|
+        @all << new(row).freeze
+      end
+      @all.freeze
+    # rescue PG::Error => e
```
Comment (iceraluk) [inline app/models/concerns/static_record.rb]: do we want commented code on here?

File: app/models/concerns/static_record.rb [hunk 2]
```diff
@@ -32,5 +32,21 @@ def where(arg)
...
+    # puts e.message
+    rescue StandardError => error
+      # print detail.backtrace.join("\n")
```
Comment (iceraluk) [inline app/models/concerns/static_record.rb]: and here

File: app/models/facilities_management/region.rb [hunk 3]
```diff
@@ -41,5 +41,6 @@ def self.all_codes
     end
   end
 
-  Region.load_csv('facilities_management/regions.csv')
+  # Region.load_csv('facilities_management/regions.csv')
```
Comment (iceraluk) [inline app/models/facilities_management/region.rb]: and here

File: app/models/nuts1_region.rb [hunk 4]
```diff
@@ -8,4 +8,5 @@ def nuts2_regions
   end
 end
 
-Nuts1Region.load_csv('nuts1_regions.csv')
+# Nuts1Region.load_csv('nuts1_regions.csv')
```
Comment (iceraluk) [inline app/models/nuts1_region.rb]: and here

File: app/models/nuts2_region.rb [hunk 5]
```diff
@@ -16,4 +16,5 @@ def self.all_codes
   end
 end
 
-Nuts2Region.load_csv('nuts2_regions.csv')
+# Nuts2Region.load_csv('nuts2_regions.csv')
```
Comment (iceraluk) [inline app/models/nuts2_region.rb]: and here

File: app/models/nuts3_region.rb [hunk 6]
```diff
@@ -10,4 +10,5 @@ def nuts2_region
   delegate :nuts1_code, :nuts1_region, to: :nuts2_region
 end
 
-Nuts3Region.load_csv('nuts3_regions.csv')
+# Nuts3Region.load_csv('nuts3_regions.csv')
```
Comment (iceraluk) [inline app/models/nuts3_region.rb]: and here

File: spec/services/static_data_loader_spec.rb [hunk 7]
```diff
@@ -0,0 +1,33 @@
+require "rails_helper"
+# require Rails.root.to_s + '/app/models/facilities_management/region.rb'
```
Comment (iceraluk) [inline spec/services/static_data_loader_spec.rb]: and here

**Comment by tariqhamid** [comment]
Please take another look. I does some reformatting to please rubocop.

**Pull request merged**